### PR TITLE
Accept "adv" key as well as "thp" key

### PR DESCRIPTION
**Pull request by mulkieran**
Related #3699

### CODE REVIEW

**Review by jbaublitz — CHANGES_REQUESTED** [review]

File: src/engine/strat_engine/backstore/backstore.rs [hunk 1]
```diff
@@ -708,19 +708,13 @@ impl Backstore {
...
-                if let Value::Object(ref mut ei) = config_to_check {
-                    ei.remove("thp");
-                }
-            }
-
-            if (existing_pin.as_str(), &config_to_check) == (pin, &parsed_config)
+            if existing_pin.as_str() == pin
+                // Check that "url" values are equal regardless of pin value.
+                // pin is tang is equivalent to config has URL.
+                && existing_info.get("url") == parsed_config.get("url")
```
[the remark this answers is not in the document]
Comment (mulkieran) [reply src/engine/strat_engine/backstore/backstore.rs]: Yes. I think we should drop that URL check altogether. I updated the code and the commit message.

**Comment by mulkieran** [comment]
@jbaublitz It's ready for you again. Please note additional source code change: https://github.com/stratis-storage/stratisd/pull/3701/commits/27cbcda5322e81b2a77c9d4a643a64f09dc554e3 .

**Comment by mulkieran** [comment]
@mvollmer If you would like to test, please let us know, thx.

**Comment by mvollmer** [comment]
> @mvollmer If you would like to test, please let us know, thx.

Will do! Thanks for the awesome overnight response!

**Comment by mvollmer** [comment]
> @mvollmer If you would like to test, please let us know, thx.

Results of manual testing, creating a encrypted Stratis pool in Cockpit bound to tang:

 - stratisd 3.6.8 from Fedora 40, works
 - stratisd from master d7e1805, fails as in #3699 
 - stratisd 3.7.1 from this PR 8160433, works

Thanks!

I'll run some of our CI tests as well.

**Review by mvollmer — APPROVED** [review]
All our tests now pass. Thanks!

**Review by jbaublitz — APPROVED** [review]
I agree that it's probably better to be a little looser with idempotence. I feel like being too specific with something as complicated as the Clevis format may lead to problems down the line.

**Pull request merged**